liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/ServerConfigDocument.java

```diff
@@ -112,7 +113,10 @@ private static void initializeAppsLocation(File serverXML, File configDir, File
                 props.putAll(propServerEnv);
             }
             
-            if (bootstrapFile.exists()) {
+            if (bootstrapProp != null && !bootstrapProp.isEmpty()) {
```

Comment:
Did you respect to the server parameters precedence? That is, bootstrip.properties in ${configDirectory} first, then `<bootstrapProperties>` in pom.xml, and  ${bootstrapPropertiesFile} last.  I don't see the condition involving ${configDirectory}.